### PR TITLE
:warning: WIP: Sign in with Github

### PR DESCRIPTION
**Pull request by sakshambhatt**
## Changes:
1. Added .env.example to indicate what env variables are being used
2. Changed the manifestplaceholders to capture the redirect url from oAuth service
3. Made the android:exported true for Android 12+
4. compile & target SdkVersions changed for AAR issue
5. Added fallback in settings.gradle for new architecture with hermes
6. Added plugins for dotenv and reanimated.
7. Added react-native-app-auth for oAuth authentication without dummy browser, react-native-keychain to store access token in secure storage, and changed react-native-reanimated version which is compatible with app's React Native version. Removed toast-message and webview as they are not used.
8. Formatting changes in FloatingActions
9. useSignIn hook has the entire logic for sign in with github which includes, authorization, calling API for code exchange, and setting states, userData, and accessToken in context, AsynStorage and secure storage respectively
10. Refactored AuthScreen to use react-native-app-auth instead of dummy browser
11. Added loading, removal of userData from AsyncStorage, and removal of access token from secure storage to logout function
12. oAuth config type declaration
13. env type declaration
14. login API call which send authcode to backend and receives access token for it from the API
15. Added removeData util to dataStore to remove a key value pair from AsyncStorage
16. renamed setSecureValue util to keychain which has set, get and removal function for react-native-keychain

## To do:
- [x] React Native Reanimated AAR meta issue
- [x] yarn issue
- [x] NDK source properties not found issue
- [x] AAR
- [x] Android 12 target
- [x] min sdk versions
- [x] Test sign in and sign out flows manually

## Test recording:
githubLogin

## Test APK link:
https://drive.google.com/file/d/1zwTlq3SrAVeTloX7sfoYH5c2U5IjLoum/view?usp=share_link

## Reviewers:
@shreya-mishra

### CODE REVIEW

File: android/app/src/main/AndroidManifest.xml
Comment (sakshambhatt) [inline android/app/src/main/AndroidManifest.xml:1]: android:exported=true added to make Android 12+ work with the SDK and latest libraries

File: android/build.gradle
Comment (sakshambhatt) [inline android/build.gradle:1]: Many libraries required min compileSdkVersion 31, so changed target and compileSdkVersions to 31

File: android/settings.gradle
Comment (sakshambhatt) [inline android/settings.gradle:1]: Partly responsible for AAR issue fix

File: babel.config.js [hunk 1]
```diff
@@ -1,4 +1,18 @@
 module.exports = {
   presets: ['module:metro-react-native-babel-preset'],
-  plugins: ['react-native-reanimated/plugin'],
+  plugins: [
+    [
```
Comment (sakshambhatt) [inline babel.config.js:4]: 1. Added safe react-native-dotenv config
2. Added react-native-reanimated config which according to its README should be placed at the bottom in all plugins

File: package.json [hunk 2]
```diff
@@ -23,19 +23,19 @@
...
     "react-native-datepicker": "^1.7.2",
     "react-native-element-dropdown": "^2.0.0",
     "react-native-floating-action": "^1.22.0",
     "react-native-gesture-handler": "^2.2.0",
     "react-native-image-picker": "^4.7.1",
+    "react-native-keychain": "^8.1.1",
     "react-native-paper": "^4.12.3",
     "react-native-radio-buttons-group": "^2.2.11",
-    "react-native-reanimated": "^2.2.4",
+    "react-native-reanimated": "^2.2.9",
```
Comment (sakshambhatt) [inline package.json:35]: 1. React Native Reanimated: Reverted to a version compatible with React Native 0.66.5
2. Added pertinent libraries

File: src/components/FloatingActions.tsx
Comment (sakshambhatt) [inline src/components/FloatingActions.tsx:1]: Rolled back the formatting change but perhaps due to minor differences still showing up

File: src/hooks/useSignIn.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+async function authorizeUser(config: Config) {
+  try {
+    const result = await authorize(config);
+    return result;
+  } catch (err) {
+    console.log(err);
+  }
+}
+
+export default function useSignIn() {
```
Comment (sakshambhatt) [inline src/hooks/useSignIn.ts:18]: Core logic for sign in with Github
1. authorization: Where a browser [Custom Tab/ SafariSafeView] opens and asks the user to sign-in to Github and then authorize the registered oAuth2 App on Github to be authorized
2. login calls the backend where the code exchange happens and client gets the accessToken, and user data
3. Access token is stored in react native keychain's secure storage, and user info in AsyncStorage

File: src/screens/AuthScreen/AuthScreen.tsx
Comment (sakshambhatt) [inline src/screens/AuthScreen/AuthScreen.tsx:1]: Relevant changes according to the new flow

File: src/types/config.d.ts
Comment (sakshambhatt) [inline src/types/config.d.ts:1]: React Native App Auth config type declaration

File: src/utils/setSecureValue.ts
Comment (sakshambhatt) [inline src/utils/setSecureValue.ts:1]: Util which stores a key value pair in react native keychain

**Comment by shreya-mishra** [comment]
We already implemented the functionality, we don't need it now!

**Pull request closed without merging**